### PR TITLE
Add swiftlint config and fix a bunch of existing linting errors. 

### PR DESCRIPTION
**Pull request by farhanpatel**
By fixing existing linting errors we'll be able to show the result of the linting in github on each pull request. We'll be able to pass/fail builds based on this which will hopefully let developers fix their linting errors saving the reviewer from having to do it. 

There are a lot of changes here. But if you work through them slowly you'll notice that they are pretty basic changes. (Tests still pass and project still builds. Soon CI will be able to tell us this!)

### CODE REVIEW

File: Storage/SQL/BrowserTable.swift [hunk 1]
```diff
@@ -166,9 +166,9 @@ public class BrowserTable: Table {
         let status = SyncStatus.New.rawValue
 
         let localArgs: Args = [
-            BookmarkRoots.RootID,    BookmarkRoots.RootGUID,          type, BookmarkRoots.RootGUID, status, now,
```
Comment (farhanpatel) [inline Storage/SQL/BrowserTable.swift]: This was a case where spaces were added to make things cleaner in the editor by allowing things to align. I'm usually not a fan of these.

File: Storage/SQL/SQLiteHistory.swift [hunk 2]
```diff
@@ -299,9 +299,7 @@ extension SQLiteHistory: BrowserHistory {
         let (query, args) = self.filteredSitesByFrecencyQueryWithHistoryLimit(limit, bookmarksLimit: 0, groupClause: groupBy, whereData: whereData)
         let cacheArgs: Args = [limit]
 
-        return accumulate([
-            { self.db.runQuery(query, args: args, factory: SQLiteHistory.iconHistoryColumnFactory) },
```
Comment (farhanpatel) [inline Storage/SQL/SQLiteHistory.swift]: A similar thing was done here. Code here was split into new lines to make them more readable.

**Comment by farhanpatel** [comment]
I've double checked all the changes. They can be categorized into 3 different kinds

1.  Remove semicolons
2. Add/Remove white space. 
3. Add/Remove new lines

**Comment by farhanpatel** [comment]
I've only turned on a limited number of rules. But there are some really good ones that will help us keep the quality of the project high. For example there are many rules that check for deprecated usages of swift. So for example `CGRectZero` is removed in favor of `CGRect.rectZero`. With swiftlint we can enable rules like this and make sure migrations are easier.

**Comment by varkor** [comment]
Is there a reason `trailing_whitespace` is disabled? As that's part of the style guide we describe under the "Whitespace" rules in readme, it seems like it would probably be a good idea to add it!

**Comment by farhanpatel** [comment]
@varkor Xcode does a good job of removing trailing whitespaces but it doesnt in comments. So lots of comments in the project have trailing spaces. I disabled the rule for now to avoid noise in this PR.

File: Storage/SQL/SQLiteHistory.swift (hunk 2 above)
Comment (sleroux) [inline Storage/SQL/SQLiteHistory.swift]: Was this throwing a linter warning? Having it all on one line makes it really long and harder to read.

File: Client/Frontend/Browser/BrowserViewController.swift [hunk 3]
```diff
@@ -1186,67 +1185,64 @@ class BrowserViewController: UIViewController {
...
         openBlankNewTabAndFocus(isPrivate: true)
     }
 
-    func closeTab(){
-        if(tabManager.tabs.count > 1){
-            tabManager.removeTab(tabManager.selectedTab!);
-        }
-        else{
+    func closeTab() {
+        if(tabManager.tabs.count > 1) {
```
Comment (sleroux) [inline Client/Frontend/Browser/BrowserViewController.swift]: Nit: We can remove these braces to clean it up further

Comment (sleroux) [inline Client/Frontend/Browser/BrowserViewController.swift]: I'm surprised that it doesn't throw a linter error for the lack of whitespace between the if and (. Probably worth just cleaning up the brackets for the methods below too.

Comment (farhanpatel) [inline Client/Frontend/Browser/BrowserViewController.swift]: There probably are rules. I've just turned many of them off for now.

Comment (farhanpatel) [inline Client/Frontend/Browser/BrowserViewController.swift]: I'm going to leave this and fix it in another PR.

Comment (varkor) [inline Client/Frontend/Browser/BrowserViewController.swift]: `operator_whitespace` for the whitespace after brackets and `control_statement` for the unnecessary brackets should do the trick, when you come back to this!
